assignment4/islandCount.py

```diff
@@ -0,0 +1,57 @@
+# Author: Li Ling
+# Last modified: 2017-06-25
+
+class IslandCounter(object):
+    """Used to count the amount of islands in a 2d landscape
+    of water and land.
+
+    Instance variables:
+        row: no.rows in the 2d array *landscape*
+        col: no.cols in the 2d array *landscape*
+        landscape: the 2d array of booleans denoting the structure of the map,
+                   True=land, False=water
+    """
+
+    def __init__(self, row, col, landscape):
+        self.row = row
+        self.col = col
+        self.landscape = landscape
+
+    def count_islands(self):
+        """ Loop through the landscape and count islands.
+        A copy of the original landscape is made and modified on the go,
+        but the original landscape is maintained. This means that the method
+        can be called multiple times and the correct result will always be obtained.
+
+        Return:
+            number of islands in the original landscape.
+        """
+        islands = 0
+        modified_landscape = self.landscape
```

Comment:
@marijja @windo You are right QAQ. The original code **does** modify the original landscape...